libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1405,6 +1408,18 @@ AP_GPS_UBLOX::_parse_gps(void)
         state.location.lng    = _buffer.pvt.lon;
         state.location.lat    = _buffer.pvt.lat;
         state.location.alt    = _buffer.pvt.h_msl / 10;
+        // Check if valid fix
+        if (!(_buffer.pvt.flags & 0b00000001) 
+            || (_buffer.pvt.fix_type != 3 && _buffer.pvt.fix_type != 4)) {
```

Comment:
the bitmasks `0b00000001`, `3` and `4` would help to not be magic numbers to provide clarity.

Reply:
I'd recommend we submit a refactor to the main ardupilot branch (if this clarify is important) so that our changes do not diverge more than necessary and add complexity to our rebase to future ardupilot updates.